### ISSUE TITLE
Local MP4 media fails to play in preview and published stories 

### ISSUE THREAD

**Issue by embarks** [GoogleForCreators/web-stories-wp#5142]
## Bug Description
MP4 videos uploaded via the media picker fail to apply the video's `src` to the `<source />` element.

## Expected Behaviour
MP4 videos uploaded via the media picker are playable in preview and publish mode.

## Steps to Reproduce
1. create a new story
2. click "upload"
3. select an MP4 video from your local files
4. insert the element into the page
5. click 'Preview'
6. console errors and "video fails to play" shows on the story

## Screenshots

![Screen Shot 2020-11-05 at 12.21.57 PM.png](https://images.zenhubusercontent.com/5f7b469a8022e02e0b23aea1/142a2436-9359-4c13-b466-fac96f2de254)
![Screen Shot 2020-11-05 at 12.22.30 PM.png](https://images.zenhubusercontent.com/5f7b469a8022e02e0b23aea1/e8d81fa9-658a-4814-88a3-39a410abaa9c)![Screen Shot 2020-11-05 at 12.22.53 PM.png](https://images.zenhubusercontent.com/5f7b469a8022e02e0b23aea1/0982fe61-d108-48e5-bf2a-b629ee93cf7b)
## Additional Context

- Plugin Version:
- Operating System: MacOS 10.15.7
- Browser: Firefox & Chrome latest

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by spacedmonkey** [comment]
Good catch. Can I strongly recommend that any fix for this contains an test of some kind. Maybe an E2E test.

**Comment by dreamofabear** [comment]
I reproduced this on two sample mp4 files on stories-lab.dev. In my case, the `amp-video` element is stripped from the rendered HTML (possibly due to AMP sanitization PHP layer).

**Comment by swissspidy** [comment]
The reason for this is a missing poster image. (see also: #5119)

If you upload a video this way and immediately click on "Preview", the poster will not have been generated yet. The markup being saved will look like this:

```html
<amp-video autoPlay="autoplay" title="video" alt="video" layout="fill" id="el-b87f132a-c34e-4f18-aa31-58e4b6356c57-media"><source type="video/mp4" src="https://example.com/wp-content/uploads/video.mp4"/></amp-video>
```

Notice the absent `poster` attriibute. And yes, that will be removed by AMP PHP sanitization due to being invalid AMP.

If you close the preview to go back to the editor, you will most likely see that the poster image has been generated by now. Click on "Preview" again to see the video in its full glory.

NB: I would say that clicking on Preview right after upload is a rare thing to do, so not that urgent.

Anyhow, I would close this as a duplicate of #5119 and work on #5189 to address this edge case.

**Comment by dreamofabear** [comment]
> If you close the preview to go back to the editor, you will most likely see that the poster image has been generated by now.
> 
> NB: I would say that clicking on Preview right after upload is a rare thing to do, so not that urgent.

This wasn't the case for me and per #5119 I don't think this is generally true.

#5119 is related but not a duplicate. "Video is missing" is significantly more severe than "story is invalid".

If this is not easily fixable, we should disable AMP sanitization in the short-term.

**Comment by swissspidy** [comment]
How easy it is to fix depends on the actual issue; as we both haven't reproduced the exact same issue as @embarks, and her output also went through AMP sanitization. Looking at the screenshot again she does have a poster image set, so it seems to be something else.
@embarks Could you perhaps test this again and record your screen? Ideally use the DevTools (https://github.com/google/web-stories-wp/blob/main/docs/devtools.md) to share your story data with us so we could try to reproduce.

We can keep the poster discussion in #5119.

**Comment by embarks** [comment]
@swissspidy Zipped up and attached recordings with DevTools on, one for the media picker and one for drag and drop.

[https://app.zenhub.com/files/235435637/e00eab54-da5d-45d9-ae00-df43800b6d24/download](https://app.zenhub.com/files/235435637/e00eab54-da5d-45d9-ae00-df43800b6d24/download)

(Zenhub does not specify the extension when you download from the link above. If you add `.zip` to the filename you'll be able to extract the recordings)

**Comment by swissspidy** [comment]
Thanks @embarks, very helpful already. Good to know it happens also with drag & drop.

(Aside: I recommended using the devtools so you could copy the data from there and paste it here :-))

I was now able to reproduce this, seeing two issues:

1. First the preview is not working because the video is still uploading, and the markup contains blob URLs.
1. Second, the AMP sanitiizer is dropping the `source` element's `src` because you're on `localhost:8899` and not using HTTPS

As per https://amp.dev/documentation/components/amp-video/, src must be HTTPS.

Will open a PR for (2)

cc @choumx

**Comment by csossi** [comment]
Verified in QA

**Comment by spacedmonkey** [comment]
Related: #5249

**Comment by bmattb** [comment]
tagged for the next release

**Comment by bmattb** [comment]
@swissspidy did this get included with 1.1.1 ?

**Comment by swissspidy** [comment]
@bmattb yes
